File: 136.java
//LeetCode 136 Single Number
// Reference: https://www.youtube.com/watch?v=KNOw_goVGas
 

class Solution {
    public int singleNumber(int[] nums) {

        //variable to store XOR value
        int sing = nums[0];

        //Loop for all elements
        for (int i = 0; i <nums.length; i++){

            //XOR the element with the previous result
            sing = sing ^ nums[i];
        }
        
        //return the result
        return sing;
    }
}

File: 136.py
# LeetCode 136: Single Number
# Reference: https://www.youtube.com/watch?v=qMPX1AOa83k

class Solution:
    def singleNumber(self, nums: List[int]) -> int:

        # Required: a linear runtime complexity and use only constant extra space.
        # XOR == Exclusive OR
        res = 0
        for n in nums:
            res = n ^ res
        return res
